.github/workflows/publish.yml

```diff
@@ -33,24 +36,38 @@ jobs:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set build tag environment variable
+        run: |
+          if [[ "${{ matrix.network }}" == "testnet" ]]; then
+            echo "BUILD_TAGS=testnet netgo" >> $GITHUB_ENV
+          else
+            echo "BUILD_TAGS=netgo" >> $GITHUB_ENV
+          fi
       - uses: docker/metadata-action@v4
-        name: generate tags
+        name: Generate tags
         id: meta
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          flavor: |
+            suffix=-${{ matrix.network }},onlatest=true
```

Comment:
I'm against adding `-mainnet` to the primary image name. I'd prefer to use conditionals.

```
name: Generate Tags Testnet
  if: matrix.network == 'testnet'
  ...
name: Generate Tags
  if: matrix.network != 'testnet'
  ...
```

Reply:
I solved it a bit differently to avoid duplicating steps. Let me know what you think

```yml
- name: Prepare environment variables
        run: |
          if [[ "${{ matrix.network }}" == "testnet" ]]; then
            echo "BUILD_TAGS=testnet netgo" >> $GITHUB_ENV
            echo "DOCKER_METADATA_suffix=-testnet" >> $GITHUB_ENV
          else
            echo "BUILD_TAGS=netgo" >> $GITHUB_ENV
            echo "DOCKER_METADATA_suffix=" >> $GITHUB_ENV
          fi
```